contracts/Staking.sol

```diff
@@ -496,32 +459,31 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         return delegation[_source][_indexer].valueAt == 0 && delegation[_source][_indexer].valueAfter == 0;
     }
 
-    function _parseStakingAmount(StakingAmount memory amount) internal view returns (uint256) {
-        IEraManager eraManager = IEraManager(settings.getEraManager());
-        uint256 eraNumber = eraManager.eraNumber();
-        if (amount.era < eraNumber) {
-            return amount.valueAfter;
-        }
-        return amount.valueAt;
+    function getTotalStakingAmount(address _indexer) external view override returns (uint256) {
+        uint256 eraNumber = IEraManager(settings.getEraManager()).eraNumber();
+        return StakingUtil.current_staking(totalStakingAmount[_indexer], eraNumber);
+    }
+
+    function getCommissionRate(address indexer) external view returns (uint256) {
+        uint256 eraNumber = IEraManager(settings.getEraManager()).eraNumber();
+        return StakingUtil.current_commission(commissionRates[indexer], eraNumber);
+    }
+
+    function getDelegationAmount(address _source, address _indexer) external view override returns (uint256) {
+        return delegation[_source][_indexer].valueAfter;
     }
 
     function getTotalEffectiveStake(address _indexer) external view override returns (uint256) {
```

Comment:
Maybe we can remove this function.
- Don't find this function use in other place.
- `effectiveStake > selfDelegation * indexerLeverageLimit` this can not happen so that no meaning for this function

Reply:
Removed, add I change `getDelegationAmount` to `getAfterDelegationAmount` for more exactly.